config/v1/types_apiserver.go

```diff
@@ -63,6 +66,31 @@ type APIServerNamedServingCert struct {
 	ServingCertificate SecretNameReference `json:"servingCertificate"`
 }
 
+type APIServerEncryption struct {
+	// type defines what encryption type should be used to encrypt resources at the datastore layer.
+	// Once this field is set to a value other than the empty string or identity, it cannot be updated.
+	// For example, setting type to aescbc permanently enables encryption of type aescbc;
+	// the type cannot be changed to a different value at a later time.
+	// When this field is unset (i.e. when it is set to the empty string), identity is implied.
+	// The behavior of unset can and will change over time.  That is, encryption will be turned
+	// on by default in a future release.  Even after encryption is enabled by default, the meaning
+	// of unset may change to a different encryption type based on changes in best practices.
+	// +optional
+	Type EncryptionType `json:"type"`
+}
```

Comment:
I want to see the encrypted resources here as defaulted string list.

Reply:
Why?  I have no desire to expose that config because it requires extra logic in the operators to understand what resources their operands own.  This is doable (I have a TODO for it), but not in scope for next release IMO.

Reply:
And in general, we know what things need encryption.  We can just do the right thing.  I would even be fine with just encrypting everything minus high write resources such as events, nodes, etc.

Reply:
I don't say to expose or even evaluate them. I say that we default them to make it clear what we support and what the default is. The reason: we do not generally encrypt. The proposed API suggests that though. With security related setting we have to be explicit. People will misunderstand.

Reply:
"We can just do the right thing" does not exist. People will have sensitive things in CRDs. For them "the right thing" is different.

Reply:
To be clear: we only allow `[ConfigMaps, Secrets]`. This will be defaulted and validated at the same time. Some day we want to add `["*"]` and maybe `excludedResources`. But that for the future. Though we should be ready to support that.